misc/drone-scripts/validate-and-serialize.sh

```diff
@@ -13,33 +13,95 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# datastore name refers to storage backed by base ESX (physical ESX) where
+# ongoing/<build_no> contains the directory named after the currently running
+# CI/ build
 DS='-ds=datastore1'
-unit_test_array=($TEST_URL_ARRAY)
-numServers=${#unit_test_array[@]}
+
+# lock names, to use in lock/unlock
+# build_lock:
+#       Overall lock makes sure only one thread at a time runs
+# check_lock:
+#       Represents a lock to perform cleanup leftover from previous run builds.
+build_lock="build_lock"
+check_lock="check_lock"
+
+# refers current CI build#
 DRONE_BUILD_NUMBER=${DRONE_BUILD_NUMBER:=0}
-prevBuildStatus=`drone build info vmware/docker-volume-vsphere $(( $DRONE_BUILD_NUMBER-$numServers ))`
-outArray=($prevBuildStatus)
 
-# Check if any other build is in the ongoing folder, if present, check if entry is stale.
-# If entry is stale clean it up.
-# If entry is an on going build wait for it.
+# wait time in seconds
+interval=30
 
-prevBuild=`govc datastore.ls $DS docker-volume-vsphere/ongoing|tail -n 1`
+# Decide whether cleanup is needed or not.
+# grab ongoing build# and parse build info, returns true when build_status
```

Comment:
nit: there is no need to repeat the code in the comment. 
Something like "returns true if the ongoing/build info is stale and the build is not running anymore, or if the user wants to restart the build so the ongoing/build and current build have the same ID"